### PR TITLE
Drive the MLIR lowering through xDSL.

### PR DESCRIPTION
**Pull request by PapyChacal**
Drive MLIR pipeline through xDSL to prepare for more back and forth.
Tested working and correct for GPU locally.

### CODE REVIEW

File: devito/core/cpu.py [hunk 1]
```diff
@@ -372,18 +373,24 @@ def _jit_compile(self):
...
+                perf("-----------------")
+                perf(f"xdsl-opt {' '.join(xdsl_args)}")
                 with redirect_stdout(out):
                     xdsl.run()
 
                 # mlir-opt
-                mlir_cmd = f'mlir-opt -p {mlir_pipeline}'
-                out = self.compile(mlir_cmd, out.getvalue())
-                #  Printer().print(out)
+                # mlir_cmd = f'mlir-opt -p {mlir_pipeline}'
```
Comment (georgebisbas) [inline devito/core/cpu.py:388]: Why would you prefer this separate?

File: devito/core/cpu.py [hunk 2]
```diff
@@ -634,19 +641,25 @@ def _jit_compile(self):
...
                 out = io.StringIO()
+                perf("-----------------")
+                perf(f"xdsl-opt {' '.join(xdsl_args)}")
                 with redirect_stdout(out):
                     xdsl.run()
 
                 # mlir-opt
-                mlir_cmd = f'mlir-opt -p {mlir_pipeline}'
-                out = self.compile(mlir_cmd, out.getvalue())
+                # mlir_cmd = f'mlir-opt -p {mlir_pipeline}'
```
Comment (georgebisbas) [inline devito/core/cpu.py:656]: same

**Review by georgebisbas — APPROVED** [review]

**Review by georgebisbas — COMMENTED** [review]
flake8 needs some fixing

**Review by georgebisbas — CHANGES_REQUESTED** [review]
flake8

**Comment by PapyChacal** [comment]
> flake8

:heavy_check_mark:

**Pull request merged**